client-js/app/elements/labrad-grapher.ts

```diff
@@ -35,30 +52,46 @@ export class LabradGrapher extends polymer.Base {
     this.showStars = false;
     this.showTrash = false;
     this.updateButtons();
+
+    this.initializeListItems_(
+      this.path,
+      this.dirs,
+      this.datasets);
+    this.applyListFilter_();
   }
 
   starClicked() {
     this.showStars = !this.showStars;
     if (this.showStars) this.showTrash = false;
     this.updateButtons();
-    this.kick++;
+    this.applyListFilter_();
   }
 
   trashClicked() {
     this.showTrash = !this.showTrash;
     if (this.showTrash) this.showStars = false;
     this.updateButtons();
-    this.kick++;
+    this.applyListFilter_();
   }
 
-  updateButtons() {
+  updateButtons(): void {
```

Comment:
Is the idea here just to have explicit return type annotations? If we're going to require that on `void` functions, we should be consistent and try to do this everywhere (for example, also on the `onKeyPress` method below). I like the idea of explicit return types on public methods, so I'd be for this.